weblate/machinery/deepl.py

```diff
@@ -60,6 +60,12 @@ def is_supported(self, source, language):
         """Check whether given language combination is supported."""
         return (source, language) in self.supported_languages
 
+    # using custom cache key to ensure that formal and informal suggestions are cached separately
+    def translate_cache_key(self, source, language, text, threshold):
+        return super().translate_cache_key(
+            source, language, text, threshold
+        ) + self.settings.get("formality", "default")
```

Comment:
Let's keep the key formatting:

```suggestion
        key = super().translate_cache_key(source, language, text, threshold)
        formality = self.settings.get("formality", "default")
        return f"{key}:{formality}"
```